### PR TITLE
E2E test improvements

### PR DESCRIPTION
**Pull request by tlrobinson**
This adds several methods similar to `@testing-library/react`'s `findBy*` methods to enzyme's `ReactWrapper`, plus allows chaining of methods that return promises, e.x. 

```javascript
  describe("browse data", () => {
    it("should load orders table and summarize", async () => {
      store.pushPath("/");
      const app = mount(store.getAppContainer());
      await app.async.findByText("Browse Data").click();
      await app.async.findByText("Sample Dataset").click();
      await app.async.findByText("Orders").click();

      await app.async.findByText("37.65");

      await app.async.findByText("Summarize").click();
      await app.async.findByText("Done").click();
      await app.async.findByText("18,760");
    });
  });
```

This this should help simplify e2e tests by reducing the dependence on implementation details.

Ideally we'd come up with a syntax that doesn't include the `.async` but the semantics of the new methods are pretty different than ezyme's default `find` methods.

### CODE REVIEW

File: frontend/test/__support__/enzyme.js [hunk 1]
```diff
@@ -155,3 +147,60 @@ export async function fillAndSubmitForm(wrapper, values) {
...
+    if (!this.__async) {
+      this.__async = {
+        find: asyncFind(this, "find"),
+        findWhere: asyncFind(this, "findWhere"),
+        findByText: asyncFind(this, "findByText"),
+        findByIcon: asyncFind(this, "findByIcon"),
+      };
+    }
+    return this.__async;
+  },
```
Comment (paulrosenzweig) [inline frontend/test/__support__/enzyme.js]: Is this getter only for lazy instantiation or does this do something else?

**Review by paulrosenzweig — APPROVED** [review]
The tests using the new functions look good!

File: frontend/test/__support__/enzyme.js (hunk 1 above)
Comment (tlrobinson) [reply frontend/test/__support__/enzyme.js]: Yes, it's lazy instantiation.

**Comment by tlrobinson** [comment]
for my benefit, remaining failures:

```
Summary of all failing tests
 FAIL  frontend/test/metabase/scenarios/query_builder.e2e.spec.js (21.357s)
  ● query builder › ask a (custom) question › should summarize and break out and show a map

    Method “type” is only meant to be run on a single node. 2 found instead.

      at ReactWrapper.single (node_modules/enzyme-2/build/ReactWrapper.js:1419:17)
      at ReactWrapper.type (node_modules/enzyme-2/build/ReactWrapper.js:1010:21)
      at click (frontend/test/__support__/enzyme.js:149:903)
      at ReactWrapper._enzyme.ReactWrapper.(anonymous function) [as click] (frontend/test/__support__/enzyme.js:182:19)
      at target.then.value (node_modules/proxymise/index.js:27:53)

 FAIL  frontend/test/metabase/admin/people/people.e2e.spec.js (6.788s)
  ● admin/people › user management › should allow admin to create new users

    The wrapper you provided for `click(wrapper)` is empty.

      at click (frontend/test/__support__/enzyme.js:149:804)
      at _callee2$ (frontend/test/metabase/admin/people/people.e2e.spec.js:79:33)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:271:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:97:21)
      at step (frontend/test/metabase/admin/people/people.e2e.spec.js:26:409)
      at frontend/test/metabase/admin/people/people.e2e.spec.js:26:569
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by tlrobinson** [comment]
Superseded by #11277 and #11278

**Pull request closed without merging**